test/policy_integration/dnspolicy_controller_test.go

```diff
@@ -137,6 +137,55 @@ var _ = Describe("DNSPolicy", func() {
 			}, TestTimeoutMedium, time.Second).Should(Succeed())
 		})
 
+		It("should not process gateway with inconsistent addresses", func() {
+			// build invalid gateway
+			gateway = testutil.NewGatewayBuilder("test-gateway", gatewayClass.Name, testNamespace).
+				WithHTTPListener(TestListenerNameOne, TestHostOne).Gateway
+			Expect(k8sClient.Create(ctx, gateway)).To(BeNil())
+
+			// ensure gateway exists
+			Eventually(func() error {
+				return k8sClient.Get(ctx, client.ObjectKey{Name: gateway.Name, Namespace: gateway.Namespace}, gateway)
```

Comment:
You can probably just do this get in the same `Eventually` closure as the status update, it's not really adding anything other than extra lines to the test file.

Reply:
That is a good advice, thanks. Joined them together